mne/cov.py

```diff
@@ -283,26 +288,19 @@ def make_ad_hoc_cov(info, verbose=None):
 
     Notes
     -----
-    This uses values of 5 fT/cm, 20 fT, and 0.2 uV for gradiometers,
+    The default noise values are 5 fT/cm, 20 fT, and 0.2 uV for gradiometers,
     magnetometers, and EEG channels, respectively.
 
     .. versionadded:: 0.9.0
     """
     picks = pick_types(info, meg=True, eeg=True, exclude=())
-
-    # Standard deviations to be used
-    grad_std = 5e-13
-    mag_std = 20e-15
-    eeg_std = 0.2e-6
-    logger.info('Using standard noise values '
-                '(MEG grad : %6.1f fT/cm MEG mag : %6.1f fT EEG : %6.1f uV)'
-                % (1e13 * grad_std, 1e15 * mag_std, 1e6 * eeg_std))
+    std = _handle_default('noise_std', std)
 
     data = np.zeros(len(picks))
     for meg, eeg, val in zip(('grad', 'mag', False), (False, False, True),
-                             (grad_std, mag_std, eeg_std)):
+                             (std['grad'], std['mag'], std['eeg'])):
         these_picks = pick_types(info, meg=meg, eeg=eeg)
-        data[np.searchsorted(picks, these_picks)] = val * val
+        data[np.searchsorted(picks, these_picks)] = np.multiply(val, val)
```

Comment:
why?

Reply:
Maybe `val` can be a list of numbers ...

Reply:
It should be a single `float` at this point in the code

Reply:
No it can still be a list of numbers. Actually an ndarray. But yeah the np.multiply is just a bad habit of mine, it's unnecessary. On the other hand, should we also allow it to be a dict of key=list of float?

Reply:
> No it can still be a list of numbers.

Why? There should be one value per channel type, which should be this `val` right?

> should we also allow it to be a dict of key=list of float?

I don't think so. For now let's just allow a single value per channel type.

Reply:
Because as it is, it can be an array, and it is really useful if one wants to implement noise with different values per sensor.

Reply:
My problem with this (as noted below) is that this is not consistent with our working definition of `ad_hoc_cov`, and how this used in MNE. For this (advanced) use case my sense is that you should construct a proper `Covariance`, as what you describe is more like a (diagonal) empirical noise cov than an ad-hoc covariance.

Reply:
I answered below :)